### DISCUSSION TITLE
Lines in the ER diagram.

### DISCUSSION THREAD

**Discussion by paoloholzl** [dbeaver/dbeaver#34139] [DBeaver Development]
Example
![schema](https://github.com/dbeaver/dbeaver/assets/96785200/6ee97b87-f40f-478c-8b2b-296afd73934e)

In the ER diagram, lines usually originate from the fields where a relationship is set (but this does not happen in DBeaver). For example, in figure 1, this is what I need. Since the schema connects the tables but not the related fields, row-by-row adjustments are required. 
It is possible to move the line, but due to the set mechanism, it is almost impossible to get from field to field without creating broken lines (see figure 2). 
Sometimes reducing the number of breaks is almost impossible.

Ideally, each line should start from the fields in referential integrity (if any), and be movable if needed. 
Then, a marker would be needed for each angle of the broken line (with the possibility of removing it with a right-click), as shown in example 3. Additionally, a way to move the line from its beginning (example 3, a marker that moves the start directly from its origin) would be helpful.

I use ER diagram programs for SQL Server, Oracle, and even MS Access, and none of them have these problems (they have other issues). I am providing an example of one of my diagrams to show that complex results can be achieved. I am also posting an image of the poor result of connecting all lines to the same origin with great difficulty. 

The tool is splendid, but this is a significant shortcoming.

Try to connect to the same filed (orrible)
![orrible](https://github.com/dbeaver/dbeaver/assets/96785200/0f2b5eee-61d7-4e9a-8b1b-c7814674b21a)

Er example
![er](https://github.com/dbeaver/dbeaver/assets/96785200/445fef1f-71f4-4b0f-8446-6f1536224d6c)

**Comment by LonwoLonwo** [comment]
Hello @paoloholzl 
What DBeaver version do you use?

**Reply by paoloholzl** [reply]
24.0.5202405191503 on Windows10

**Reply by serjiokov** [reply]
Hello @paoloholzl,
The association between attributes supported in the PRO product version known as "OrthoRouting" you can try in the trial version.
https://dbeaver.com/docs/dbeaver/ER-Diagrams/
The association Shortest Path is default and designed for connecting entire entities.

**Reply by serjiokov** [reply]
Hi, one more point use reset layout to return to default settings for diagram. Please consider positioning of items dependent on association between. 
Just a note the simplest short path doesn't support attribute association, please try commercial product version for it.

**Reply by paoloholzl** [reply]
I understand that the pro version has the function I need. 
I accept the limitation that for the Community version one has to move relationships 'by hand'.
But the functions that are in the Community version must work, manually moving the relationships is practically bordering on unusable and it is not possible to obtain an acceptable result.
In certain cases, inextricable 'knots' are created, making it impossible to move the relation line to the point where it joins the frame.
This is currently the most serious limitation I have found in the Community version.

**Reply by E1izabeth** [reply]
If you have a clear example with steps to reproduce, you can create an issue

**Reply by paoloholzl** [reply]
Ok.
For example I have soma related tables.
I create a ER diagram
![Result](https://github.com/dbeaver/dbeaver/assets/96785200/baaf5c37-2743-49cd-b5bf-59a42ad1bcfb)
but i require to correct so
![Need](https://github.com/dbeaver/dbeaver/assets/96785200/2b9833d6-831e-4eea-8c52-73514cb5f7a7)
Now I try ... impossible to do

https://github.com/dbeaver/dbeaver/assets/96785200/c962524c-0cf8-4bcc-8508-05a245f512b4

**Reply by E1izabeth** [reply]
Thank you!

**Reply by E1izabeth** [reply]
I created the issue https://github.com/dbeaver/dbeaver/issues/34649
